lib/multi.c

```diff
@@ -2019,6 +2019,19 @@ static CURLMcode multi_runsingle(struct Curl_multi *multi,
       break;
 
     case MSTATE_DO:
+      if(data->set.fprereq) {
+        /* call the prerequest callback function */
+        Curl_set_in_callback(data, true);
+        data->set.fprereq(data->set.prereq_userp,
```

Comment:
For most callbacks we want to allow the callback to abort the transfer with a special return code. This one as well I'm sure.

Reply:
I can certainly look into it if you think it's something that's necessary.

Reply:
Added this.